7.1 

Duotas skaicius n, po jo n eilučių iš porų (žodis, skaičius). Parašykite programą, kuri naudojant lambda funkciją surikiuoja rinkinių sąrašą marks pagal antrą reikšmę mažėjimo tvarka.

4
English 10
Science 9
Maths 8
Social sciences 10


[(’English’, 10), (’Social sciences’, 10), (’Science’, 9), (’Maths’, 8)]

In [31]:
n = int(input())
d = []
for i in range(n):
    a = input().split()
    d.append((' '.join(a[:-1]), int(a[-1])))

d.sort(key=lambda x: x[1], reverse = True)
print(d)

ValueError: invalid literal for int() with base 10: '{"a":500, "b":5874, "c": 560, "d":400, "e":5874, "f": 20}'

7.2

Duotas sąrašas L. Funkcijos lambda pagalba parašykite programą, kuri sukuria sąrašo L elementų kubų sąrašą.

[12,10,25,2,17,3,5]
[1728, 1000, 15625, 8, 4913, 27, 125]

In [ ]:
L = eval(input())
print(list(map(lambda x: x**3, L)))

[1728, 1000, 15625, 8, 4913, 27, 125]


7.3

Duotas sąrašas L. Parašykite programą, kuri surikiuoja sąrašo elementus pagal antrą raidę.

["manner","unfair","release","free","fixture","user","emphasis","coach","character","skilled","spoil","equinox"]

In [ ]:
L = eval(input())
L.sort(key=lambda x: x[1])
print(L)

['manner', 'release', 'character', 'fixture', 'skilled', 'emphasis', 'unfair', 'coach', 'spoil', 'equinox', 'free', 'user']


7.4

Duoti du žodynai. Apjunkite juos į vieną žodyną ir išveskite rezultatą didėjimo tvarka.

{"Ten": 10, "Twenty": 20, "Thirty": 30}
{"Thirty": 30, "Fourty": 40, "Fifty": 50}

Ten - 10
Twenty - 20
Thirty - 30
Fourty - 40
Fifty - 50


In [ ]:
d1 = dict(eval(input()))
d2 = dict(eval(input()))
d1.update(d2)

for x,y in d1.items():
    print(x, '-', y)

Ten - 10
Twenty - 20
Thirty - 30
Fourty - 40
Fifty - 50


7.5

Duota: sąrašas empl ir žodynas defaults. Metodo dict.fromkeys() pagalba sudarykite žodyną d. Išveskite žodyno d duomenis stulpeliu kaip parodyta.

["Kelly", "John", "Emma"]
{"designation": "Application Developer", "salary": 8000}

Kelly: {’designation’: ’Application Developer’, ’salary’: 8000}
John: {’designation’: ’Application Developer’, ’salary’: 8000}
Emma: {’designation’: ’Application Developer’, ’salary’: 8000}

In [ ]:
empl = eval(input())
defaults = eval(input())
d = dict.fromkeys(empl, defaults)
for key, value in d.items():
    print(key+':', value)

Kelly: {'designation': 'Application Developer', 'salary': 8000}
John: {'designation': 'Application Developer', 'salary': 8000}
Emma: {'designation': 'Application Developer', 'salary': 8000}


7.6

Iš duotų aibių keys ir values sudarykite žodyną funkcijos zip() pagalba.

("Tokyo", "Delhi", "Shanghai", "Mumbai")
(38140000, 26454000, 24484000)

{’Tokyo’: 38140000, ’Delhi’: 26454000, ’Shanghai’: 24484000}

In [ ]:
keys = eval(input())
values = eval(input())
d = dict(zip(keys, values))
print(d)

{'Tokyo': 38140000, 'Delhi': 26454000, 'Shanghai': 24484000}


7.7

Parašykite programą, kuri iš ekrano nuskaito žodyną ir randa jo reikšmių sumą.

{"x": 100, "b": 100, "e": 200, "a": 300, "h": 100, "d": 100}            900

In [ ]:
d = eval(input())
print(sum(d.values()))

900


7.8

Duotas žodynas, vardas ir x reikšmė. Pakeiskite duoto vardo "salary" reikšmę į x.

{"emp1": {"name": "John", "salary": 8500},"emp2": {"name": "Navy", "salary": 6000},"emp3": {"name": "Brad", "salary": 7500}}
Navy
8000

In [ ]:
d = eval(input())
vardas = input()
x = int(input())

for i in d.keys():
    if d[i]['name'] == vardas:
        d[i]['salary'] = x

print(d)

{'emp1': {'name': 'John', 'salary': 8500}, 'emp2': {'name': 'Navy', 'salary': 8000}, 'emp3': {'name': 'Brad', 'salary': 7500}}


7.9

Duotas žodynas d. Išveskite dviejų porų su mažiausiomis reikšmėmis sąrašą.

{"a":500, "b":5874, "c": 560, "d":400, "e":5874, "f": 20}

[(’f’, 20), (’d’, 400)]

In [36]:
d = eval(input())
d = dict(L)

d_sort = sorted(d.items(), key=lambda x: (x[1], x[0]))
d_sort = sorted(d.items(), key=lambda x: x[1])[:2]
print(d_sort)

[('a', 3), ('c', 3)]
